### PR TITLE
Handle numeric code values in assert_throws

### PR DESCRIPTION
**Pull request by lukebjerring**
Allows a `number` type to be passed in, as specified in the method's documentation.

### CODE REVIEW

File: resources/testharness.js [hunk 1]
```diff
@@ -1436,11 +1436,28 @@ policies and contribution forms [3].
...
+                }
             }
 
-            var required_props = { code: name_code_map[name] };
+            var required_props = { code };
+
+            if (typeof code === "number") {
+                if (code === 0) {
+                    throw new AssertionError('Test bug: ambiguous DOMException code 0 passed to assert_throws()');
+                } else if (!Array.from(Object.values(name_code_map)).includes(code)) {
```
Comment (lukebjerring) [inline resources/testharness.js]:
```suggestion
                } else if (!(code in code_name_map)) {
```

File: resources/test/tox.ini [hunk 2]
```diff
@@ -14,4 +14,4 @@ deps =
   six
   requests
 
-commands = pytest -vv {posargs}
+commands = pytest -vv {posargs} tests/functional/api-tests-1.html
```
Comment (gsnedders) [inline resources/test/tox.ini]: This will make it only run this file.

File: resources/testharness.js [hunk 3]
```diff
@@ -1436,11 +1436,28 @@ policies and contribution forms [3].
                 NotAllowedError: 0
             };
 
-            if (!(name in name_code_map)) {
-                throw new AssertionError('Test bug: unrecognized DOMException code "' + code + '" passed to assert_throws()');
+            var code_name_map = {};
+            for (let [k, v] of Object.entries(name_code_map)) {
```
Comment (gsnedders) [inline resources/testharness.js]: Let's keep testharness.js working in more older browsers if we can and just rely on ES5 syntax.

**Review by gsnedders — DISMISSED** [review]

File: resources/testharness.js (hunk 3 above)
Comment (lukebjerring) [reply resources/testharness.js]: Done. If that's the long-term intention (stay es5), might be good to get a lint set up.

File: resources/test/tox.ini (hunk 2 above)
Comment (lukebjerring) [reply resources/test/tox.ini]: Reverted - was a local change to speed up my testing.

File: resources/testharness.js (hunk 3 above)
Comment (jugglinmike) [reply resources/testharness.js]: [And also documented.](https://github.com/web-platform-tests/wpt/pull/6333)

File: resources/testharness.js [hunk 4]
```diff
@@ -1436,11 +1436,28 @@ policies and contribution forms [3].
                 NotAllowedError: 0
             };
 
-            if (!(name in name_code_map)) {
-                throw new AssertionError('Test bug: unrecognized DOMException code "' + code + '" passed to assert_throws()');
-            }
+            var code_name_map = {};
+            Object.entries(name_code_map).forEach(function(keyValue) {
```
Comment (jugglinmike) [inline resources/testharness.js]: Even though we're primarily concerned with new language syntax, the use of new built-ins like `Object.entries` could be a problem when used in something as generic (and old) as `assert_throws`. I'm more conservative than most about this, though, and it passed @gsnedders' review, so I'm open to suggestion.

Comment (lukebjerring) [reply resources/testharness.js]: Done.

File: resources/testharness.js [hunk 5]
```diff
@@ -1436,11 +1436,28 @@ policies and contribution forms [3].
...
+                if (keyValue[1] > 0) {
+                    code_name_map[keyValue[1]] = keyValue[0];
+                }
+            });
+
+            var required_props = { code };
```
[the remark this answers is not in the document]
Comment (lukebjerring) [reply resources/testharness.js]: Done.

**Review by jugglinmike — APPROVED** [review]

**Pull request merged**